cmd/cluster/core/create_test.go

```diff
@@ -0,0 +1,89 @@
+package core
+
+import (
+	"context"
+	"testing"
+
+	. "github.com/onsi/gomega"
+
+	apiversion "k8s.io/apimachinery/pkg/version"
+	fakediscovery "k8s.io/client-go/discovery/fake"
+	fakekubeclient "k8s.io/client-go/kubernetes/fake"
+)
+
+func TestValidateMgmtClusterAndNodePoolCPUArchitectures(t *testing.T) {
+	ctx := context.Background()
+
+	fakeKubeClient := fakekubeclient.NewSimpleClientset()
+	fakeDiscovery, ok := fakeKubeClient.Discovery().(*fakediscovery.FakeDiscovery)
+
+	if !ok {
+		t.Fatalf("failed to convert FakeDiscovery")
+	}
+
+	// if you want to fake a specific version
+	fakeDiscovery.FakedServerVersion = &apiversion.Info{
+		Platform: "linux/amd64",
+	}
+
+	tests := []struct {
+		name        string
+		opts        *RawCreateOptions
+		expected    bool
+		expectError bool
+	}{
+		{
+			name: "Test case 1: Valid multi-arch image",
+			opts: &RawCreateOptions{
+				ReleaseImage:   "quay.io/openshift-release-dev/ocp-release:4.16.13-multi",
+				PullSecretFile: "../../../hack/dev/fakePullSecret.json",
+				ReleaseStream:  "4-stable",
+				Arch:           "amd64",
+			},
+			expectError: false,
+		},
+		{
+			name: "Test case 2: No release image provided but a multi-arch release stream is provided",
+			opts: &RawCreateOptions{
+				ReleaseImage:   "",
+				PullSecretFile: "../../../hack/dev/fakePullSecret.json",
+				ReleaseStream:  "4-stable-multi",
+				Arch:           "amd64",
+			},
+			expectError: false,
+		},
+		{
+			name: "Test case 3: Invalid multi-arch image (release stream not provided or release image is invalid but mgmt cpu and nodepool arches match",
+			opts: &RawCreateOptions{
+				ReleaseImage:   "quay.io/openshift-release-dev/ocp-release:4.16.13-x86_64",
+				PullSecretFile: "../../../hack/dev/fakePullSecret.json",
+				ReleaseStream:  "4-stable",
+				Arch:           "amd64",
+			},
+			expectError: false,
+		},
+		{
+			name: "Test case 4: Release image and release stream are valid but mgmt & NodePool CPU architectures do not match but mgmt cpu and nodepool arches do not match",
+			opts: &RawCreateOptions{
+				ReleaseImage:   "quay.io/openshift-release-dev/ocp-release:4.16.13-x86_64",
+				PullSecretFile: "../../../hack/dev/fakePullSecret.json",
+				ReleaseStream:  "4-stable",
+				Arch:           "arm64",
+			},
+			expectError: true,
+		},
+	}
+
+	for _, tc := range tests {
+		t.Run(tc.name, func(t *testing.T) {
+			g := NewWithT(t)
+			t.Setenv("FAKE_CLIENT", "true")
```

Comment:
why do you need to set this env variable? you decoupled the client now

Reply:
This was left behind while I was debugging the unit tests. This has been removed now.